pkg/publish.go

```diff
@@ -97,3 +124,104 @@ func Publish(file string, rootID string, dryRun bool) (id string, err error) {
 	}
 	return c.ID, nil
 }
+
+func GetImageSrcsFromHTML(content string) []string {
+	var imageSrcs []string
+	z := html.NewTokenizer(strings.NewReader(content))
+	for {
+		tt := z.Next()
+		t := z.Token()
+		switch tt {
+		case html.ErrorToken:
+			return imageSrcs
+		case html.SelfClosingTagToken:
+			if t.Data == "img" {
+				for _, attr := range t.Attr {
+					if attr.Key == "src" {
+						imageSrcs = append(imageSrcs, attr.Val)
+					}
+				}
+			}
+		}
+	}
+	return imageSrcs
+}
+
+func GetImageSrcFiles(content string, file string) []string {
+	fileDir := filepath.Dir(file)
+	imageSrcs := GetImageSrcsFromHTML(content)
+
+	var imageSrcFiles []string
+
+	for _, imageSrc := range imageSrcs {
+		imageSrcPath := filepath.Join(fileDir, imageSrc)
+		if _, err := os.Stat(imageSrcPath); !os.IsNotExist(err) {
+			imageSrcFiles = append(imageSrcFiles, imageSrc)
+		}
+	}
+
+	return imageSrcFiles
+}
+
+func ReplaceImagesWithAttachments(imageSrcFiles []string, file string, pageContent string, pageID string, wiki *confluence.Wiki, uri string) (string, error) {
+	for _, imageSrcFile := range imageSrcFiles {
+		fileDir := filepath.Dir(file)
+		imageSrcPath := filepath.Join(fileDir, imageSrcFile)
+		imageSrcFilename := filepath.Base(imageSrcPath)
+
+		results, err := wiki.GetAttachment(pageID, imageSrcFilename)
+		if err != nil {
+			return "", err
+		}
+
+		if len(results.Results) == 0 {
+			// create new attachment
+			_, err := wiki.CreateAttachment(pageID, imageSrcPath)
+			if err != nil {
+				return "", err
+			}
+		} else {
+			// update existing attachment
+			imageData, err := wiki.GetAttachmentData(pageID, imageSrcFilename)
+			if err != nil {
+				return "", err
+			}
+
+			attachmentSum := getBytesSha256(imageData)
+			fileSum, err := getFileSha256(imageSrcPath)
+			if err != nil {
+				return "", err
+			}
+
+			if fileSum != attachmentSum {
+				_, err := wiki.UpdateAttachment(pageID, imageSrcPath, results.Results[0].ID)
+				if err != nil {
+					return "", err
+				}
+			}
+		}
+
+		pageContent = strings.Replace(pageContent, "\"" + imageSrcFile + "\"", "\"" + uri + "/download/attachments/" + pageID + "/" + imageSrcFilename + "\"", -1)
```

Comment:
Consider using back-tick strings to enhance readability when they contain double-quote literals.
```
fmt.Sprintf(`"%s"`, imageSrcFile)
```

Reply:
- [ ] use backticks for strings with `"`